web/job_runs_controller.go

```diff
@@ -22,23 +23,51 @@ type JobRunsController struct {
 
 // Index returns paginated JobRuns for a given JobSpec
 // Example:
-//  "<application>/specs/:SpecID/runs?size=1&page=2"
+//  "<application>/runs?jobSpecId=:jobSpecId&size=1&page=2"
 func (jrc *JobRunsController) Index(c *gin.Context) {
-	id := c.Param("SpecID")
+	id := c.Query("jobSpecId")
 	size, page, offset, err := ParsePaginatedRequest(c.Query("size"), c.Query("page"))
 	if err != nil {
 		c.AbortWithError(422, err)
 		return
 	}
-	var jrs []models.JobRun
-	if count, err := jrc.App.GetStore().JobRunsCountFor(id); err != nil {
-		c.AbortWithError(500, fmt.Errorf("error getting count of JobRuns: %+v", err))
-	} else if err := jrc.App.GetStore().Find("JobID", id, &jrs, storm.Limit(size), storm.Skip(offset), storm.Reverse()); err != nil {
-		c.AbortWithError(500, fmt.Errorf("error getting JobRuns: %+v", err))
-	} else if buffer, err := NewPaginatedResponse(*c.Request.URL, size, page, count, jrs); err != nil {
-		c.AbortWithError(500, fmt.Errorf("failed to marshal document: %+v", err))
+
+	var order func(opts *index.Options)
+	if c.Query("sort") == "-createdAt" {
+		order = storm.Reverse()
 	} else {
-		c.Data(200, MediaType, buffer)
+		order = func(opts *index.Options) {}
+	}
+
+	limit := storm.Limit(size)
+	skip := storm.Skip(offset)
+
+	var runs []models.JobRun
+
+	if id == "" {
```

Comment:
* Assuming this is acting as a filter for the jobSpecId

Rather than have this large `if/else`, let's break them out into their respective methods with self documenting names:

```golang
if specId == "" {
  handleAllJobRuns(c)
} else {
  handleJobRunsFor(specId, c)
}
```

Inside each method, you could pull out the metadata and DRY it up in a method:

```golang
func handleAllJobRuns(...) {
  reqParams, err := pullOutParams(c)
  ...
  } else if err := jrc.App.GetStore().All(&runs, reqParams.limit, reqParams.skip, reqParams.order); err != nil {
```

spitballing here.

Reply:
@dimroc updated. I also changed the queries so it can share error handling logic.